### DISCUSSION TITLE
Major changes to everything! (v0.4.0)

### DISCUSSION THREAD

**Discussion by Mostlime12195** [WebOsu-2/webosu-2.github.io#20] [Announcements]
Our previous update still feels fresh, but I've been extra motivated recently and have been wanting to do this update for a while.

First off, with our biggest change (whether you notice it or not), is that we updated our graphics/input/audio library! That's right, we updated it from the ancient 5.2.0 to the slightly-less-ancient 6.5.10. Yes, it's still a couple years old, and still 2 full major updates old (current version of PixiJS is v8.4.2, as of writing this, eesh.)

The change might not sound like much, but it's a huge step into (eventually) modernising the code. A modernised and up-to-date source code can aid immensely for future development and compatibility.

Another change, which might actually be a little more noticable, is the replacement of XHR for fetch! Before, the user (you!) downloaded everything using the outdated, unrecommended, and simply terrible XHR with the cleaner and simpler fetch api. With fetch, you may receive fewer errors and issues while downloading beatmaps, and that ghastly glitch with users that have Internet Download Manager installed should be fixed.

Finally, a couple site layout changes have been made, the most obvious of which being that the footer is now centred.

Happy Christmas!

**Comment by Mellowfow** [comment]
Glad this is still being developed

**Reply by Mostlime12195** [reply]
I'm trying my best, although javascript isn't my favourite language to work with and I have lots of school work.
